x/gamm/pool-models/stableswap/amm.go

```diff
@@ -221,7 +237,7 @@ func solveCFMMBinarySearchMulti(xReserve, yReserve, wSumSquares, yIn osmomath.Bi
 		panic(err)
 	}
 
-	xOut := xReserve.Sub(xEst)
+	xOut := xReserve.Sub(xEst).Abs()
```

Comment:
Err, are we sure this abs is correct

Reply:
Yes – it's there to cover the case where `yIn` is negative and `xEst > xReserve`

Alternatively, we can have an if/else that does `xOut = xReserve - xEst` or `xOut = xEst -  xReserve` but this seemed cleaner to me

Reply:
I definitely think this abs should be done by the caller.

We are equivocating two different things in the function output with the abs (caller needs to give tokens into the pool, vs caller gets tokens out of the pool)